netcore/Singletons/AMQPClientSingleton.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Security.Authentication;
+using Amqp;
+
+namespace OrderCaptureAPI.Singetons
+{
+    public sealed class AMQPClientSingleton : IDisposable
+    {
+        private static readonly AMQPClientSingleton instance = new AMQPClientSingleton();
+        private static volatile SenderLink _senderLinkInstance;
+        private static volatile Address _amqpAddress;
+        private static volatile Connection _amqpConnection;
+        private static volatile Session _amqpSession;
+        private static volatile string _amqpHost;
+        private static volatile string _eventHubEntity;
+
+        public static string AMQPHost
+        {
+            get
+            {
+                return _amqpHost;
+            }
+        }
+
+        // Explicit static constructor to tell C# compiler
+        // not to mark type as beforefieldinit
+        static AMQPClientSingleton()
+        {
+        }
+
+        private AMQPClientSingleton()
+        {
+            // Retrieve the AMQPHost from the Environment Variables
+            _amqpHost = System.Environment.GetEnvironmentVariable("AMQPHOST");
```

Comment:
Will need to update the docs with the environment variable name. I opted to simplify by having a common name AMQPHOST to work with both RabbitMQ and Event Hub.